### DISCUSSION TITLE
Finding Top-K Matches For A Subset of Subsequence Patterns or Motifs

### DISCUSSION THREAD

**Discussion by lqmike** [TDAmeritrade/stumpy#349] [Help: Other Questions]
Can matrix profile return not only the best match but also second best, third best etc.. at this moment Matrix Profile by definition only record the minimum value i.e the best match of a given pattern, but in the case where there are more than 1 best match, i.e the 2nd and 3rd in the list isn't far off, I would like to to see those patterns as well as confirmation. Thank you!

**Comment by seanlaw** [comment]
@lqmike So, it sounds like what you are asking for is a "top-K" match for each pattern. 99% of the time, the user is really only looking for the top-K matches for a few select subsequences and rarely are they interested in recording the top-K matches for EVERY subsequence within their time series. Thus, the typical process is to:

1. Compute the matrix profile
2. From the matrix profile, identify the subset of motifs/subsequence patterns that you are interested in. 
3. Using this subset of subsequences, compute the distance profile via the [fast pattern searching approach](https://stumpy.readthedocs.io/en/latest/Tutorial_Pattern_Searching.html)

Let me know if that helps

**Reply by lqmike** [reply]
Thank you Sean for the super fast reply. Okay I understand your suggestion now. The reason for "EVERY subsequence” is that I am running sort of backtest, so looping and recording over all the historical data to see how effective the predictability of similar patterns.
